### ISSUE TITLE
Applying scopes leads to errors when running query with include.hierarchy = true

### ISSUE THREAD

**Issue by vbogdanov** [overlookmotel/sequelize-hierarchy#15]
```javascript
HierarchyModel.scope('any').findOne({
				where: {
					id: 123
				},
				include: [
					{
						model: HierarchyModel,
						as: 'descendents',
						hierarchy: true
					}
				]
			});
```
results in
```SequelizeHierarchyError: You cannot get a hierarchy of 'HierarchyModel' without including it from a parent```
Commenting the scope call is enough to remove the error. After apply the contents of the scope to my query (one more include) the query ran successfully giving the expected result. The reason for this behavior seems to be:
https://github.com/overlookmotel/sequelize-hierarchy/blob/master/lib/hooksUniversal.js#L99
```
if (includeModel !== model) throw new Sequelize.SequelizeHierarchyError(...
```
as Model.scope('...') results in a new object, that is not equal to the previous one. As a result this check fails unless I'm using the same model reference in the hierarchy definition.
```javascript
let tmp = HierarchyModel.scope('withgrants');
tmp.findOne({
				where: {
					id: 123
				},
				include: [
					 {
						model: tmp,
						as: 'descendents',
						hierarchy: true
					}
				]
			});
```
works just fine and 
```javascript
let tmp = HierarchyModel.scope('withgrants');
HierarchyModel.scope('withgrants').findOne({
				where: {
					id: 123
				},
				include: [
					 {
						model: HierarchyModel.scope('withgrants'),
						as: 'descendents',
						hierarchy: true
					}
				]
			});
```
does not.
Would you please change the check from referential equality (includeModel !== model) to model name or table name or some other acceptable identifier so that I can use scopes in both places.

**Comment by overlookmotel** [comment]
Hi @vbogdanov.

I'm not really up to speed with how scopes work - I don't use them myself. Just to give me an idea, can you please post the definition of your scope(s)?

What I'm wondering is whether the included model should also be subject to the scope or not. I mean this part:

```js
{
    model: HierarchyModel,
    as: 'descendents',
    hierarchy: true
}
```

Also, I'm afraid I don't really have time to implement this right now. But would be very happy to receive a PR (with appropriate tests), if you'd be up for it... looks like you've figured out how it needs to be changed to work.

**Comment by overlookmotel** [comment]
By the way, thanks very much for the detailed analysis of your issue @vbogdanov.

**Comment by vbogdanov** [comment]
definition (names changed):
```javascript
scopes: {
	withgrants: function () {
		return {
			include: [
				{
					model: HierarchyModelGrant,
					as: 'HierarchyModelGrants'
				}
			]
		};
	}
}
```
where
```javascript
HierarchyModelGrant.belongsTo(HierarchyModel)
HierarchyModel.hasMany(HierarchyModelGrant)
```

**Comment by overlookmotel** [comment]
So would you want the scope to apply to the `descendents`?

I guess if so this should be specified with:

```js
HierarchyModel.scope('withgrants').findOne({
    where: { id: 123 },
    include: [ {
        model: HierarchyModel.scope('withgrants'),
        as: 'descendents',
        hierarchy: true
    } ]
});
```

I mean rather than the hierarchy include being automatically scoped the same as the main model.

If so, then yes you're right that simply checking the model names against each other rather than `if (includeModel !== model) ...` would solve the problem.

Would you have time to submit a PR for this? I'll merge it immediately if you can provide a test too.

**Comment by vbogdanov** [comment]
If you merge it please publish a new version in npm also.

**Comment by overlookmotel** [comment]
Fixed in PR #16.
